### ISSUE TITLE
>1.1.2 doesn't compile in TypeScript projects with `dom` library disabled

### ISSUE THREAD

**Issue by shevchenkobn** [pulsecron/pulse#40]
### Description

1. Have a TypeScript project/codebase **without** `dom` type definitions (it can be done through setting `tsc --lib es6` or similar).
2. Install version after _(not equal)_ 1.1.2 (including the latest - 1.5.1).
3. Import anything from this package.
4. Compile the project.

Compilation fails with the error:

```
node_modules/@pulsecron/pulse/dist/job/unique.d.ts:3:42 - error TS2304: Cannot find name 'Document'.

3 export type UniqueMethod<TSchema extends Document = Document> = (filter: Filter<TSchema>, options?: {
                                           ~~~~~~~~

node_modules/@pulsecron/pulse/dist/job/unique.d.ts:3:53 - error TS2304: Cannot find name 'Document'.

3 export type UniqueMethod<TSchema extends Document = Document> = (filter: Filter<TSchema>, options?: {
```

##### `index.ts`

```typescript

// import {Document as MongoDocument} from 'mongodb';

// declare module '@pulsecron/pulse/dist/job/unique' {
//   type Document = MongoDocument;
// }

import {Processor} from '@pulsecron/pulse';

const p: Processor<any> = () => {};
```

##### `package.json`

```json
{
  "name": "ts",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build": "tsc --types node --lib es6 ./index.ts",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@pulsecron/pulse": "^1.5.1"
  },
  "devDependencies": {
    "@types/node": "^20.14.9",
    "mongodb": "^6.8.0",
    "typescript": "^5.5.2"
  }
}

```

#### Workaround

Add the declaration augmentation file (like `src\@types\pulse-fixes.d.ts`):

```typescript
import {Document as MongoDocument} from 'mongodb';

declare module '@pulsecron/pulse/dist/job/unique' {
  type Document = MongoDocument;
}
```

Or simply uncomment the commented part of the **Code example**.

#### Fix

Import `Document` from `mongodb` in `@pulsecron/pulse/src/job/unique` 😀 https://github.com/pulsecron/pulse/blob/dfed3ee042dd1fc63e3aff74f6bdf4032c3e561c/src/job/unique.ts#L1

And disable DOM type definitions by [`lib` in `tsconfig`](https://www.typescriptlang.org/tsconfig/lib.html).

**Comment by code-xhyun** [comment]
@shevchenkobn  I will solve these and give you an answer as soon as possible.

**Comment by code-xhyun** [comment]
🎉 This Issue is included in version 1.5.2 🎉

The release is available on:

[npm package (@latest dist-tag)](https://www.npmjs.com/package/@pulsecron/pulse/v/1.5.2)
[GitHub release](https://github.com/pulsecron/pulse/releases/tag/v1.5.2)
Your [semantic-release](https://github.com/semantic-release/semantic-release) bot 📦🚀

**Comment by shevchenkobn** [comment]
> 🎉 This Issue is included in version 1.5.2 🎉
> 
> The release is available on:
> 
> [npm package (@latest dist-tag)](https://www.npmjs.com/package/@pulsecron/pulse/v/1.5.2) [GitHub release](https://github.com/pulsecron/pulse/releases/tag/v1.5.2)

Wow, thank you! You did it blazingly fast! Even I took much longer to respond 😄
I have never seen such a fast open-source issue processing!

**Comment by code-xhyun** [comment]
> > 🎉 This Issue is included in version 1.5.2 🎉
> 
> > 
> 
> > The release is available on:
> 
> > 
> 
> > [npm package (@latest dist-tag)](https://www.npmjs.com/package/@pulsecron/pulse/v/1.5.2) [GitHub release](https://github.com/pulsecron/pulse/releases/tag/v1.5.2)
> 
> 
> 
> Wow, thank you! You did it blazingly fast! Even I took much longer to respond 😄
> 
> I have never seen such a fast open-source issue processing!

@shevchenkobn  I've solved this, but please check if your problem has been solved as well
